### PR TITLE
Allow sub-apps to be mounted on the root path

### PR DESCRIPTION
**Pull request by track0x1**
Express.js allows you to mount sub-applications to the root path (aka. it doesn't require a separate entry path). This change hoists routes from a provided Polka instance to the primary instance.

### CODE REVIEW

**Comment by lukeed** [comment]
Hey, thanks~!

I'm trying to understand the point or benefit of this approach. Do you mind explaining your reasoning for this?

Also, let's discuss it first before addressing the notes I've made 😄 Don't want you to spend more time on this until I'm sold -- because as of now I don't think I find this to be useful.

**Comment by mram** [comment]
This might probably work
``` js
let loop = _ => res.finished || (i < len)
  ? ((arr[i+1] instanceof Polka) ? arr[i++].handler : arr[i++])(req, res, next);
  : fn(req, res);

// or

let loop = _ => res.finished || (i < len)
  ? (((arr[i+1] instanceof Polka)
     ? arr[i++].handler.bind(null, req, res)
     : arr[i++].bind(req, res, next))
  )()
  : fn(req, res);
```
for
``` js
// index.js
const polka = require('polka')();
polka.use(require('./routes/core'));
polka.use(require('./routes/pages'));
polka.listen();

// routes/core.js
const polka = require('polka')();
polka.get('/ping', (req, res) => res.end('pong'));
module.exports = polka;

// routes/pages.js
const polka = require('polka')();
polka.get('/page-one', (req, res) => res.end('my page'));
module.exports = polka;
```

**Comment by lukeed** [comment]
@mram We can't do expensive/computationally-heavier logic in the main handler. Either of your snippets could (very potentially) dramatically decrease performance.

@track0x1 I'm shooting for familiarity, not exact parity. 

This still feels like almost like an anti-pattern to me, to be honest. I'm still leaning towards having this live in a user-land solution. Something like:

```js
// my-polka.js
const polka = require('polka')

module.exports = () => {
  let app = polka();
  
  app.attach = sub => {
    let m, keys, obj=sub.handlers;
    for (m in obj) {
      if ((keys=Object.keys(obj[m])).length) {
        keys.forEach(uri => app.add(m, uri, obj[m][uri]));
      }
    }
    app.wares = app.wares.concat(sub.wares);
    app.bwares = app.bwares.concat(sub.bwares);
  }

  return app;
}
```

Then using it:

```js
const polka = require('./my-polka');

polka()
  .attach(require('./routes/core'))
  .attach(require('./routes/pages'))
  .listen();
```

**Comment by mram** [comment]
> We can't do expensive/computationally-heavier logic in the main handler. Either of your snippets could (very potentially) dramatically decrease performance.

Second thought, realised it 😬

**Comment by lukeed** [comment]
@mram Compare with the results in the README or in the benchmarks directory~

**Snippet 1**

```
Running 10s test @ http://localhost:3000
  8 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     2.43ms  293.78us   7.65ms   91.27%
    Req/Sec     4.96k     1.22k   39.05k    99.88%
  395250 requests in 10.10s, 39.20MB read
Requests/sec:  39135.96
Transfer/sec:      3.88MB
```

**Snippet 2** (modified)

```js
// modified to make it work
let loop = _ => res.finished || (i < len)
  ? (
    arr[i+1] instanceof Polka
      ? arr[i++].handler.call(null, req, res)
      : arr[i++](req, res, next)
  ) : fn(req, res);
```

```
Running 10s test @ http://localhost:3000
  8 threads and 100 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     2.42ms  228.76us   7.67ms   88.38%
    Req/Sec     4.99k   367.95     9.85k    97.26%
  398964 requests in 10.10s, 39.57MB read
Requests/sec:  39495.00
Transfer/sec:      3.92MB
```

Important to note that this is still running the same benchmark code, a simple `GET /`, which represents the best-case scenario. There are no sub-application mounted. Any sub-applications with these approaches would produce far worse results, since that's where the extra work comes in.

Basically, these benchmarks are showing the cost of adding an extra `arr[i+1] instanceof Polka` within each loop cycle.

**Comment by track0x1** [comment]
@lukeed Been thinking this over; what do you think about a dependency injection approach to handle this?

```js
// index.js
const polka = require('polka')();
require('./routes/core')(polka);
require('./routes/pages')(polka);
polka.listen();

// routes/core.js
module.exports = (polka) => {
  polka.get('/ping', (req, res) => res.end('pong'));
};

// routes/pages.js
module.exports = (polka) => {
  polka.get('/page-one', (req, res) => res.end('my page'));
};
```
